ci/integration.sh

```diff
@@ -39,41 +36,25 @@ function check_fmt {
     if [[ $? != 0 ]]; then
         return 1
     fi
-    cargo test --all
-    if [[ $? != 0 ]]; then
-        return $?
-    fi
-}
-
-function check {
-    cargo test --all
-    if [[ $? != 0 ]]; then
-        return 1
-    fi
-    check_fmt
-    if [[ $? != 0 ]]; then
-        return 1
-    fi
```

Comment:
Note that `stdsimd` was broken because the doc tests failed to build after a reformat. IMO we should be checking that if the library builds correctly before formatting, it should also build correctly afterwards, and that applies to all its tests, including doc tests.

Reply:
Ah, I see. I'll add that back in once I've worked out what is going on with the rustfmt failures

Reply:
Many of the failures are either warnings because the crates need pull request to remove deprecated options, or because rustfmt emits warnings or errors that the user cannot do anything about. 

If there is nothing that the user can do about a warning or internal error, then the diagnostic should IMO be opt-in (e.g. behind a `rustfmt --debug-output` or similar).

I've tried to enable these diagnostics here because even if they are opt-in, they should be fixed at some point. However a "red" allowed to fail travis build bot because of a warning can hide a change that actually breaks the build so... I did not know what the right way to proceed was here and did not wanted to add all buildbots twice...

Reply:
What's puzzling me is that some of the errors *are* meant to be opt-in with `--error-on-unformatted` and when I run locally I don't see the errors, but running on CI is doing something different to my local setup.

Reply:
I figured it out! The directories here are nested under rustfmt, whereas my local tests were not, that means they see rustfmt's rustfmt.toml which opts into the debug mode stuff. Testing now without that...